src/app/dataset-view/dataset.component.ts

```diff
@@ -281,8 +283,19 @@ export class DatasetComponent extends BaseDatasetDataComponent implements OnInit
     }
 
     public onRunSQLRequest(params: DatasetRequestBySql): void {
+        const startTime = new Date().valueOf();
+        this.sqlLoading = true;
```

Comment:
Unit test this logic with successful and failing queries.
Use `fakeAsync` and `tick` to control exact time flow.